lms/djangoapps/courseware/module_render.py

```diff
@@ -1002,11 +1008,37 @@ def handle_xblock_callback(request, course_id, usage_id, handler, suffix=None):
         suffix (str)
 
     Raises:
+        HttpResponseForbidden: If the request method is not in SAFE_METHODS and user is not authenticated.
         Http404: If the course is not found in the modulestore.
     """
-    # NOTE (CCB): Allow anonymous GET calls (e.g. for transcripts). Modifying this view is simpler than updating
-    # the XBlocks to use `handle_xblock_callback_noauth`...which is practically identical to this view.
-    if request.method != 'GET' and not request.user.is_authenticated:
+    # In this case, we are using Session based authentication, so we need to check CSRF token.
+    if request.user.is_authenticated:
+        error = CsrfViewMiddleware().process_view(request, None, (), {})
+        if error:
+            return error
+
+    # We are reusing DRF logic to provide support for JWT and Oauth2. We abandoned the idea of using DRF view here
+    # to avoid introducing backwards-incompatible changes.
+    # You can see https://github.com/edx/XBlock/pull/383 for more details.
+    else:
+        authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser)
+        authenticators = [auth() for auth in authentication_classes]
+
+        for authenticator in authenticators:
+            try:
+                user_auth_tuple = authenticator.authenticate(request)
+            except APIException:
+                log.exception(
+                    "XBlock handler %r failed to authenticate with %s", handler, authenticator.__class__.__name__
+                )
+            else:
+                if user_auth_tuple is not None:
+                    request.user, _ = user_auth_tuple
+                    break
+
+    # NOTE (CCB): Allow anonymous SAFE_METHOD calls (e.g. for transcripts). Modifying this view is simpler than updating
+    # the XBlocks to use `handle_xblock_callback_noauth`, which is practically identical to this view.
+    if not (request.method in SAFE_METHODS or request.user and request.user.is_authenticated):
```

Comment:
@Agrendalath, using parenthesis to group the condition will make it more readable.

```suggestion
    if not (request.method in SAFE_METHODS or (request.user and request.user.is_authenticated)):
```